src/analysis/multiSpecies/Microbiome_Modeling_Toolbox/PairwiseInteractionModeling/simulatePairwiseInteractions.m

```diff
@@ -0,0 +1,375 @@
+%
+% This script predicts the outcomes of pairwise simulations in every
+% combination from a given list of pairwise models. The pairwise models
+% need to be created first with the function joinModelsPairwiseFromList.
+% This script requires the COBRA Toolbox function solveCobraLP. Due to the
+% coupling constraints on the pairwise models, the simulations cannot
+% currently be run with optimizeCbModel.
+% Below is a description of all possible consequences for the two joined
+% organisms. Please note that the outcomes depend on the two genome-scale
+% reconstructions joined and are highly dependent on the applied
+% constraints.
+%
+% * Competition: both organisms grow slower in co-growth than separately
+%   (same outcome for both).
+%
+% * Parasitism: one organism grows faster in co-growth than separately, while
+%   the other grows slower in co-growth than separately. Outcome for
+%   faster-growing organism: Parasitism_Taker, for slower-growing organism:
+%   Parasitism_Giver
+%
+% * Amensalism: one organism's growth is unaffected by co-growth,  while
+%   the other grows slower in co-growth than separately. Outcome for
+%   unaffected organism: Amensalism_Unaffected, for slower-growing organism:
+%   Amensalism_Affected
+%
+% * Neutralism: both organisms' growths are unaffected by co-growth (same
+%   outcome for both)
+%
+% * Commensalism: one organism's growth is unaffected by co-growth,  while
+%   the other grows fatser in co-growth than separately. Outcome for
+%   unaffected organism: Commensalism_Giver, for slower-growing organism:
+%   Commensalism_Taker
+%
+% * Mutualism: both organisms growth faster in co-growth than separately
+%   (same outcome for both)
+%
+% INPUTS
+% inputModels          Array of pairwise model structures to be simulated
+% pairedModelInfo      Information on species joined in the pairwise models
+%
+% OPTIONAL INPUTS
+% inputDiet            Cell array of three columns containing exchange reaction
+%                      model abbreviations, lower bounds, and upper bounds. 
+%                      If no diet is input then the input pairwise models 
+%                      will be used with unchanged constraints.
+% saveSolutionsFlag    If true, flux solutions are stored (may result in 
+%                      large output files)
+% numWorkers           Number of workers in parallel pool if desired
+% sigD                 Difference in growth rate that counts as significant
+%                      change (by default: 10%)
+% OUTPUTS
+% pairwiseInteractions     Table with computed pairwise and single growth
+%                          rates for all entered microbe-microbe models
+% OPTIONAL OUTPUT
+% pairwiseSolutions        Table with all computed flux solutions saved
+%                         (may result in large output files)
+%
+% .. Authors:
+%       - Almut Heinken, 02/2018 
+
+% if no diet was entered
+if nargin <2
```

Comment:
please use `if ~exist('argumentName','var')` or an `inputParser` and not nargin < X for readability and maintainability.